osu.Game/Graphics/UserInterface/OsuModSettingsNumberBox.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class OsuModSettingsNumberBox : OsuModSettingsTextBox
```

Comment:
A more tolerable way I would consider for this is making a `SettingsNumberBox` in the same fashion as `SettingsTextBox` and expose the nested number box class from inside it (which would be inheriting the one in `SettingsTextBox`) publicly for use in `ModRandom`.